src/cryptonote_basic/cryptonote_basic.h

```diff
@@ -225,8 +225,10 @@ namespace cryptonote
 
     transaction();
     transaction(const transaction &t);
+    transaction(transaction &&t);
     transaction &operator=(const transaction &t);
-    virtual ~transaction();
```

Comment:
Nothing inherits from transaction. So just this change alone will reduce the size of the struct by 8 bytes (assuming 64-bit system). Given millions of transactions, that's a lot of memory! 👍